app/menus/TemplatesMenu.tsx

```diff
@@ -25,36 +25,83 @@ function TemplatesMenu({ onSelectTemplate, document }: Props) {
   const user = useCurrentUser();
   const { documents } = useStores();
   const { t } = useTranslation();
-  const templates = documents.templates;
 
-  if (!templates.length) {
-    return null;
-  }
+  const templateToMenuItem = React.useCallback(
+    (tmpl: Document): MenuItem => ({
+      type: "button",
+      title: replaceTitleVariables(tmpl.titleWithDefault, user),
+      icon: tmpl.icon ? (
+        <Icon value={tmpl.icon} color={tmpl.color ?? undefined} />
+      ) : (
+        <DocumentIcon />
+      ),
+      onClick: () => onSelectTemplate(tmpl),
+    }),
+    [user, onSelectTemplate]
+  );
+
+  const templates = documents.templates.filter((tmpl) => tmpl.publishedAt);
+
+  const workspaceTemplates = templates
+    .filter((tmpl) => tmpl.isWorkspaceTemplate)
+    .map(templateToMenuItem);
+
+  const templatesInCollection = templates
+    .filter(
+      (tmpl) =>
+        !tmpl.isWorkspaceTemplate && tmpl.collectionId === document.collectionId
+    )
+    .map(templateToMenuItem);
 
-  const templatesInCollection = templates.filter(
-    (t) => t.collectionId === document.collectionId
+  const otherTemplates = templates
+    .filter(
+      (tmpl) =>
+        !tmpl.isWorkspaceTemplate && tmpl.collectionId !== document.collectionId
+    )
+    .map(templateToMenuItem);
+
+  const workspaceItems: MenuItem[] = React.useMemo(
+    () =>
+      workspaceTemplates.length
+        ? [{ type: "heading", title: t("Workspace") }, ...workspaceTemplates]
+        : [],
+    [t, workspaceTemplates]
   );
-  const otherTemplates = templates.filter(
-    (t) => t.collectionId !== document.collectionId
+
+  const collectionItemsWithHeader: MenuItem[] = React.useMemo(
+    () =>
+      templatesInCollection.length
+        ? [
+            { type: "separator" },
+            { type: "heading", title: t("This collection") },
+            ...templatesInCollection,
+          ]
+        : [],
+    [t, templatesInCollection]
   );
 
-  const renderTemplate = (template: Document) => (
-    <MenuItem
-      key={template.id}
-      onClick={() => onSelectTemplate(template)}
-      icon={
-        template.icon ? (
-          <Icon value={template.icon} color={template.color ?? undefined} />
-        ) : (
-          <DocumentIcon />
-        )
-      }
-      {...menu}
-    >
-      {replaceTitleVariables(template.titleWithDefault, user)}
-    </MenuItem>
+  const otherItemsWithHeader: MenuItem[] = React.useMemo(
+    () =>
+      otherTemplates.length
+        ? [
+            { type: "separator" },
+            { type: "heading", title: t("Other collections") },
```

Comment:
I think as part of this we should just remove this section, once you can share templates across collections at the workspace level it becomes unnecessary. This menu starts to look pretty messy with three headers. With this section gone it could be:

```
Collection templates
-- Workspace templates header --
Workspace templates
```